### PR TITLE
Adding magic table to check for libmagic data

### PR DESCRIPTION
**Pull request by javuto**
Now osquery comes with magic!

Magic

### CODE REVIEW

File: osquery/tables/system/magic.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genMagicData(QueryContext& context) {
+  QueryData results;
+  magic_t magic_cookie;
```
Comment (theopolis) [inline osquery/tables/system/magic.cpp]: Like @marpaia pointed out: this is actually an opaque pointer type:
```cpp
typedef struct magic_set *magic_t;
```

If this is initialized: `magic_t magic_cookie = nullptr;` we'll benefit from static analysis.

File: osquery/tables/system/magic.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  // No default flags
+  magic_cookie = magic_open(MAGIC_NONE);
+
+  if (!magic_cookie) {
```
Comment (theopolis) [inline osquery/tables/system/magic.cpp]: Compare this to a `nullptr`: `if (magic_cookie != nullptr) {`

File: osquery/tables/system/magic.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    VLOG(1) << "Unable to initialize magic library.";
+    return results;
+  }
+  if (magic_load(magic_cookie, nullptr) != 0) {
+    VLOG(1) << "Unable to load magic database - " << magic_error(magic_cookie);
```
Comment (theopolis) [inline osquery/tables/system/magic.cpp]: nit, change `-` to a `:` for logging consistency.

File: osquery/tables/system/magic.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+QueryData genMagicData(QueryContext& context) {
+  QueryData results;
+  magic_t magic_cookie;
+
+  // No default flags
+  magic_cookie = magic_open(MAGIC_NONE);
+
+  if (!magic_cookie) {
+    VLOG(1) << "Unable to initialize magic library.";
```
Comment (theopolis) [inline osquery/tables/system/magic.cpp]: nit, remove the `.` here.

FYI most logs don't end in `.` for some reason? 
```
» grep -rn "LOG(" . | grep "\.\"" | wc -l
      15
» grep -rn "LOG(" . | wc -l
     190
```

Those 15 have arrest warrants posted.

**Comment by theopolis** [comment]
tswiftlgtm

**Comment by javuto** [comment]
4054c9dfdce4b53c094010582496c2bb8e7a8944d3315bff05dd7327b4c7a3e3

**Pull request merged**